## 1 - Banco de Dados

In [1]:
import pandas as pd
import numpy as np

In [2]:
path = 'C:/Users/Akihiro/Desktop/TERA/Módulo_2/Deesafio de Inferência Estatística'
phq = pd.read_csv(path+'/DEMO_PHQ.csv', sep=',', encoding='UTF-8', header=0, names=['ID', 'Q01', 'Q02', 'Q03', 'Q04', 'Q05', 'Q06',
       'Q07', 'Q08', 'Q09', 'Gênero', 'Idade', 'Etnia',
       'Escolaridade', 'Renda_familiar'])
pag_hei = pd.read_csv(path+'/PAG_HEI.csv', sep=',', encoding='UTF-8', header=0, names=['ID', 'PAG_MINW', 'ADHERENCE', 'HEI2015C1_TOTALVEG',
                                                                                       'HEI2015C2_GREEN_AND_BEAN', 'HEI2015C3_TOTALFRUIT',
                                                                                       'HEI2015C4_WHOLEFRUIT', 'HEI2015C5_WHOLEGRAIN', 'HEI2015C6_TOTALDAIRY',
                                                                                       'HEI2015C7_TOTPROT', 'HEI2015C8_SEAPLANT_PROT', 'HEI2015C9_FATTYACID',
                                                                                       'HEI2015C10_SODIUM', 'HEI2015C11_REFINEDGRAIN', 'HEI2015C12_SFAT',
                                                                                       'HEI2015C13_ADDSUG', 'HEI2015_TOTAL_SCORE'])

## 2 - Tratamento de Dados

### Preencher valores nulos  do Demo_phq.csv

In [3]:
phq.head()

ID  Q01  Q02  Q03  Q04  Q05  Q06  Q07  Q08  Q09  Gênero  Idade  Etnia  \
0  31130  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN  NaN       2     85      3   
1  31131  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0       2     44      4   
2  31132  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0       1     70      3   
3  31134  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0       1     73      3   
4  31139  0.0  0.0  0.0  0.0  3.0  1.0  0.0  0.0  0.0       2     18      2   

   Escolaridade  Renda_familiar  
0             4             4.0  
1             4            11.0  
2             5            11.0  
3             3            12.0  
4             3            11.0

In [4]:
phq.tail()

ID  Q01  Q02  Q03  Q04  Q05  Q06  Q07  Q08  Q09  Gênero  Idade  \
5329  41466  0.0  0.0  1.0  1.0  0.0  0.0  0.0  0.0  0.0       2     58   
5330  41468  0.0  2.0  0.0  1.0  1.0  2.0  1.0  3.0  0.0       2     66   
5331  41469  1.0  0.0  0.0  1.0  0.0  0.0  0.0  0.0  0.0       1     19   
5332  41472  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0       1     34   
5333  41473  0.0  0.0  2.0  1.0  0.0  1.0  0.0  0.0  0.0       1     21   

      Etnia  Escolaridade  Renda_familiar  
5329      5             2             3.0  
5330      1             1             8.0  
5331      4             4             2.0  
5332      3             4             7.0  
5333      1             2             6.0

Valores nulos estão presentes na tabela phq. Verificar os valores nulos gerais

In [5]:
phq.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5334 entries, 0 to 5333
Data columns (total 15 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   ID              5334 non-null   int64  
 1   Q01             4836 non-null   float64
 2   Q02             4836 non-null   float64
 3   Q03             4836 non-null   float64
 4   Q04             4836 non-null   float64
 5   Q05             4836 non-null   float64
 6   Q06             4836 non-null   float64
 7   Q07             4836 non-null   float64
 8   Q08             4836 non-null   float64
 9   Q09             4836 non-null   float64
 10  Gênero          5334 non-null   int64  
 11  Idade           5334 non-null   int64  
 12  Etnia           5334 non-null   int64  
 13  Escolaridade    5334 non-null   int64  
 14  Renda_familiar  5287 non-null   float64
dtypes: float64(10), int64(5)
memory usage: 625.2 KB


Q01 até Q09: cada um com 498 valores nulos, o que é relevante para a análise.

In [6]:
phq.describe()

ID          Q01          Q02          Q03          Q04  \
count   5334.000000  4836.000000  4836.000000  4836.000000  4836.000000   
mean   36358.796775     0.307279     0.313069     0.551282     0.717328   
std     2964.731827     0.758459     0.710683     0.931166     0.913711   
min    31130.000000     0.000000     0.000000     0.000000     0.000000   
25%    33838.500000     0.000000     0.000000     0.000000     0.000000   
50%    36347.500000     0.000000     0.000000     0.000000     0.000000   
75%    38915.750000     0.000000     0.000000     1.000000     1.000000   
max    41473.000000     9.000000     9.000000     9.000000     9.000000   

               Q05          Q06          Q07          Q08          Q09  \
count  4836.000000  4836.000000  4836.000000  4836.000000  4836.000000   
mean      0.320720     0.209884     0.230356     0.147849     0.050248   
std       0.743057     0.624552     0.639186     0.590390     0.326933   
min       0.000000     0.000000     0.000000     0.000000     0.000000   
25%       0.000000     0.000000     0.000000     0.000000     0.000000   
50%       0.000000     0.000000     0.000000     0.000000     0.000000   
75%       0.000000     0.000000     0.000000     0.000000     0.000000   
max       9.000000     9.000000     9.000000     9.000000     9.000000   

            Gênero        Idade        Etnia  Escolaridade  Renda_familiar  
count  5334.000000  5334.000000  5334.000000   5334.000000     5287.000000  
mean      1.519873    45.089989     2.871016      3.233221        8.738604  
std       0.499652    20.145853     1.128468      1.268395       12.777756  
min       1.000000    18.000000     1.000000      1.000000        1.000000  
25%       1.000000    27.000000     3.000000      2.000000        4.000000  
50%       2.000000    43.000000     3.000000      3.000000        7.000000  
75%       2.000000    62.000000     4.000000      4.000000       10.000000  
max       2.000000    85.000000     5.000000      9.000000       99.000000

In [7]:
print('Gênero:', phq[phq['Q02'].isnull()]['Gênero'].value_counts())
print('Etnia:', phq[phq['Q02'].isnull()]['Etnia'].value_counts())

Gênero: 2    268
1    230
Name: Gênero, dtype: int64
Etnia: 3    180
4    152
1    111
5     37
2     18
Name: Etnia, dtype: int64


Gênero feminino e etnia mexino-americana são fatores importantes para imputar os dados de Q02. Farei uma classificação de idade, tornando-a mais eficiente.

In [8]:
age_cat_edges = [0, 3, 12, 18, 35, 60, 1000]
phq['Categoria_idade'] = pd.cut(phq['Idade'], 
                                age_cat_edges, 
                                labels=['Bebê', 'Criança', 'Adolescente',
                                       'Jovem Adulto', 'Adulto', 'Idoso'])

In [9]:
# Q02
print('Categoria_idade:', phq[phq['Q02'].isnull()]['Categoria_idade'].value_counts())
# Mulher jovem adulta:
print('Média para mulher jovem adulta:', phq[phq['Q02'].notnull()][phq['Categoria_idade']=='Jovem Adulto'][phq['Gênero']==2]['Q02'].mean())
phq_02_media_pmja = phq[phq['Q02'].notnull()][phq['Categoria_idade']=='Jovem Adulto'][phq['Gênero']==2]['Q02'].mean()
phq[phq['Categoria_idade']=='Jovem Adulto'][phq['Gênero']==2]['Q02'] = phq['Q02'].fillna(phq_02_media_pmja, inplace=True)
# Homem jovem adulto:
print('Média para homem jovem adulto:', phq[phq['Q02'].notnull()][phq['Categoria_idade']=='Jovem Adulto'][phq['Gênero']==1]['Q02'].mean())
phq_02_media_phja = phq[phq['Q02'].notnull()][phq['Categoria_idade']=='Jovem Adulto'][phq['Gênero']==1]['Q02'].mean()
phq[phq['Categoria_idade']=='Jovem Adulto'][phq['Gênero']==1]['Q03'] = phq['Q02'].fillna(phq_02_media_phja, inplace=True)
# Mulher adulta:
print('Média para mulher adulta:', phq[phq['Q02'].notnull()][phq['Categoria_idade']=='Adulto'][phq['Gênero']==2]['Q02'].mean())
phq_02_media_pma = phq[phq['Q02'].notnull()][phq['Categoria_idade']=='Adulto'][phq['Gênero']==2]['Q02'].mean()
phq[phq['Categoria_idade']=='Adulto'][phq['Gênero']==2]['Q02'] = phq['Q02'].fillna(phq_02_media_pma, inplace=True)
# Homem adulto:
print('Média para homem adulto:', phq[phq['Q02'].notnull()][phq['Categoria_idade']=='Adulto'][phq['Gênero']==1]['Q02'].mean())
phq_02_media_pha = phq[phq['Q02'].notnull()][phq['Categoria_idade']=='Adulto'][phq['Gênero']==1]['Q02'].mean()
phq[phq['Categoria_idade']=='Adulto'][phq['Gênero']==1]['Q02'] = phq['Q02'].fillna(phq_02_media_pha, inplace=True)

Categoria_idade: Jovem Adulto    171
Adulto          155
Idoso           144
Adolescente      28
Bebê              0
Criança           0
Name: Categoria_idade, dtype: int64
Média para mulher jovem adulta: 0.3110871905274489
Média para homem jovem adulto: 0.28903018722028884
Média para mulher adulta: 0.4205333342806842
Média para homem adulto: 0.28410241931214575


C:\Users\Akihiro\AppData\Local\Temp/ipykernel_16380/3579812703.py:4: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  print('Média para mulher jovem adulta:', phq[phq['Q02'].notnull()][phq['Categoria_idade']=='Jovem Adulto'][phq['Gênero']==2]['Q02'].mean())
C:\Users\Akihiro\AppData\Local\Temp/ipykernel_16380/3579812703.py:5: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  phq_02_media_pmja = phq[phq['Q02'].notnull()][phq['Categoria_idade']=='Jovem Adulto'][phq['Gênero']==2]['Q02'].mean()
C:\Users\Akihiro\AppData\Local\Temp/ipykernel_16380/3579812703.py:6: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  phq[phq['Categoria_idade']=='Jovem Adulto'][phq['Gênero']==2]['Q02'] = phq['Q02'].fillna(phq_02_media_pmja, inplace=True)
C:\Users\Akihiro\AppData\Local\Temp/ipykernel_16380/3579812703.py:8: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  print('Média para homem jovem ad

In [10]:
phq.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5334 entries, 0 to 5333
Data columns (total 16 columns):
 #   Column           Non-Null Count  Dtype   
---  ------           --------------  -----   
 0   ID               5334 non-null   int64   
 1   Q01              4836 non-null   float64 
 2   Q02              5334 non-null   float64 
 3   Q03              4836 non-null   float64 
 4   Q04              4836 non-null   float64 
 5   Q05              4836 non-null   float64 
 6   Q06              4836 non-null   float64 
 7   Q07              4836 non-null   float64 
 8   Q08              4836 non-null   float64 
 9   Q09              4836 non-null   float64 
 10  Gênero           5334 non-null   int64   
 11  Idade            5334 non-null   int64   
 12  Etnia            5334 non-null   int64   
 13  Escolaridade     5334 non-null   int64   
 14  Renda_familiar   5287 non-null   float64 
 15  Categoria_idade  5334 non-null   category
dtypes: category(1), float64(10), int64(5)
memo

In [11]:
# Q03
print('Gênero:', phq[phq['Q03'].isnull()]['Gênero'].value_counts())
print('Etnia:', phq[phq['Q03'].isnull()]['Etnia'].value_counts())
print('Categoria idade:', phq[phq['Q03'].isnull()]['Categoria_idade'].value_counts())
# Criar variável por gênero e idade:
# Mulher jovem adulta:
print('Média para mulher jovem adulta:', phq[phq['Q03'].notnull()][phq['Categoria_idade']=='Jovem Adulto'][phq['Gênero']==2]['Q03'].mean())
phq_03_media_pmja = phq[phq['Q03'].notnull()][phq['Categoria_idade']=='Jovem Adulto'][phq['Gênero']==2]['Q03'].mean()
phq[phq['Categoria_idade']=='Jovem Adulto'][phq['Gênero']==2]['Q03'] = phq['Q03'].fillna(phq_03_media_pmja, inplace=True)
# Homem jovem adulto:
print('Média para homem jovem adulto:', phq[phq['Q03'].notnull()][phq['Categoria_idade']=='Jovem Adulto'][phq['Gênero']==1]['Q03'].mean())
phq_03_media_phja = phq[phq['Q03'].notnull()][phq['Categoria_idade']=='Jovem Adulto'][phq['Gênero']==1]['Q03'].mean()
phq[phq['Categoria_idade']=='Jovem Adulto'][phq['Gênero']==1]['Q03'] = phq['Q03'].fillna(phq_03_media_phja, inplace=True)
# Mulher adulta:
print('Média para mulher adulta:', phq[phq['Q03'].notnull()][phq['Categoria_idade']=='Adulto'][phq['Gênero']==2]['Q03'].mean())
phq_03_media_pma = phq[phq['Q03'].notnull()][phq['Categoria_idade']=='Adulto'][phq['Gênero']==2]['Q03'].mean()
phq[phq['Categoria_idade']=='Adulto'][phq['Gênero']==2]['Q03'] = phq['Q03'].fillna(phq_03_media_pma, inplace=True)
# Homem adulto:
print('Média para homem adulto:', phq[phq['Q03'].notnull()][phq['Categoria_idade']=='Adulto'][phq['Gênero']==1]['Q03'].mean())
phq_03_media_pha = phq[phq['Q03'].notnull()][phq['Categoria_idade']=='Adulto'][phq['Gênero']==1]['Q03'].mean()
phq[phq['Categoria_idade']=='Adulto'][phq['Gênero']==1]['Q03'] = phq['Q03'].fillna(phq_03_media_pha, inplace=True)

Gênero: 2    268
1    230
Name: Gênero, dtype: int64
Etnia: 3    180
4    152
1    111
5     37
2     18
Name: Etnia, dtype: int64
Categoria idade: Jovem Adulto    171
Adulto          155
Idoso           144
Adolescente      28
Bebê              0
Criança           0
Name: Categoria_idade, dtype: int64
Média para mulher jovem adulta: 0.6221743810548978
Média para homem jovem adulto: 0.45661334601680525
Média para mulher adulta: 0.6463471966141718
Média para homem adulto: 0.5277676801543462


C:\Users\Akihiro\AppData\Local\Temp/ipykernel_16380/2639721855.py:7: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  print('Média para mulher jovem adulta:', phq[phq['Q03'].notnull()][phq['Categoria_idade']=='Jovem Adulto'][phq['Gênero']==2]['Q03'].mean())
C:\Users\Akihiro\AppData\Local\Temp/ipykernel_16380/2639721855.py:8: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  phq_03_media_pmja = phq[phq['Q03'].notnull()][phq['Categoria_idade']=='Jovem Adulto'][phq['Gênero']==2]['Q03'].mean()
C:\Users\Akihiro\AppData\Local\Temp/ipykernel_16380/2639721855.py:9: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  phq[phq['Categoria_idade']=='Jovem Adulto'][phq['Gênero']==2]['Q03'] = phq['Q03'].fillna(phq_03_media_pmja, inplace=True)
C:\Users\Akihiro\AppData\Local\Temp/ipykernel_16380/2639721855.py:11: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  print('Média para homem jovem a

In [12]:
# Q04
print('Gênero:', phq[phq['Q04'].isnull()]['Gênero'].value_counts())
print('Etnia:', phq[phq['Q04'].isnull()]['Etnia'].value_counts())
print('Categoria idade:', phq[phq['Q04'].isnull()]['Categoria_idade'].value_counts())
# Criar variável por gênero e idade:
# Mulher jovem adulta:
print('Média para mulher jovem adulta:', phq[phq['Q04'].notnull()][phq['Categoria_idade']=='Jovem Adulto'][phq['Gênero']==2]['Q04'].mean())
phq_04_media_pmja = phq[phq['Q04'].notnull()][phq['Categoria_idade']=='Jovem Adulto'][phq['Gênero']==2]['Q04'].mean()
phq[phq['Categoria_idade']=='Jovem Adulto'][phq['Gênero']==2]['Q04'] = phq['Q04'].fillna(phq_04_media_pmja, inplace=True)
# Homem jovem adulto:
print('Média para homem jovem adulto:', phq[phq['Q04'].notnull()][phq['Categoria_idade']=='Jovem Adulto'][phq['Gênero']==1]['Q04'].mean())
phq_04_media_phja = phq[phq['Q04'].notnull()][phq['Categoria_idade']=='Jovem Adulto'][phq['Gênero']==1]['Q04'].mean()
phq[phq['Categoria_idade']=='Jovem Adulto'][phq['Gênero']==1]['Q04'] = phq['Q04'].fillna(phq_04_media_phja, inplace=True)
# Mulher adulta:
print('Média para mulher adulta:', phq[phq['Q04'].notnull()][phq['Categoria_idade']=='Adulto'][phq['Gênero']==2]['Q04'].mean())
phq_04_media_pma = phq[phq['Q04'].notnull()][phq['Categoria_idade']=='Adulto'][phq['Gênero']==2]['Q04'].mean()
phq[phq['Categoria_idade']=='Adulto'][phq['Gênero']==2]['Q04'] = phq['Q04'].fillna(phq_04_media_pma, inplace=True)
# Homem adulto:
print('Média para homem adulto:', phq[phq['Q04'].notnull()][phq['Categoria_idade']=='Adulto'][phq['Gênero']==1]['Q04'].mean())
phq_04_media_pha = phq[phq['Q04'].notnull()][phq['Categoria_idade']=='Adulto'][phq['Gênero']==1]['Q04'].mean()
phq[phq['Categoria_idade']=='Adulto'][phq['Gênero']==1]['Q04'] = phq['Q04'].fillna(phq_04_media_pha, inplace=True)

Gênero: 2    268
1    230
Name: Gênero, dtype: int64
Etnia: 3    180
4    152
1    111
5     37
2     18
Name: Etnia, dtype: int64
Categoria idade: Jovem Adulto    171
Adulto          155
Idoso           144
Adolescente      28
Bebê              0
Criança           0
Name: Categoria_idade, dtype: int64
Média para mulher jovem adulta: 0.8740581270182992
Média para homem jovem adulto: 0.5772690481554729
Média para mulher adulta: 0.8659807853767084


C:\Users\Akihiro\AppData\Local\Temp/ipykernel_16380/19613708.py:7: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  print('Média para mulher jovem adulta:', phq[phq['Q04'].notnull()][phq['Categoria_idade']=='Jovem Adulto'][phq['Gênero']==2]['Q04'].mean())
C:\Users\Akihiro\AppData\Local\Temp/ipykernel_16380/19613708.py:8: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  phq_04_media_pmja = phq[phq['Q04'].notnull()][phq['Categoria_idade']=='Jovem Adulto'][phq['Gênero']==2]['Q04'].mean()
C:\Users\Akihiro\AppData\Local\Temp/ipykernel_16380/19613708.py:9: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  phq[phq['Categoria_idade']=='Jovem Adulto'][phq['Gênero']==2]['Q04'] = phq['Q04'].fillna(phq_04_media_pmja, inplace=True)
C:\Users\Akihiro\AppData\Local\Temp/ipykernel_16380/19613708.py:11: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  print('Média para homem jovem adulto:',

Média para homem adulto: 0.6210956019458028


C:\Users\Akihiro\AppData\Local\Temp/ipykernel_16380/19613708.py:20: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  phq_04_media_pha = phq[phq['Q04'].notnull()][phq['Categoria_idade']=='Adulto'][phq['Gênero']==1]['Q04'].mean()
C:\Users\Akihiro\AppData\Local\Temp/ipykernel_16380/19613708.py:21: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  phq[phq['Categoria_idade']=='Adulto'][phq['Gênero']==1]['Q04'] = phq['Q04'].fillna(phq_04_media_pha, inplace=True)


In [13]:
# Q05
print('Gênero:', phq[phq['Q05'].isnull()]['Gênero'].value_counts())
print('Etnia:', phq[phq['Q05'].isnull()]['Etnia'].value_counts())
print('Categoria idade:', phq[phq['Q05'].isnull()]['Categoria_idade'].value_counts())
# Criar variável por gênero e idade:
# Mulher jovem adulta:
print('Média para mulher jovem adulta:', phq[phq['Q05'].notnull()][phq['Categoria_idade']=='Jovem Adulto'][phq['Gênero']==2]['Q05'].mean())
phq_05_media_pmja = phq[phq['Q05'].notnull()][phq['Categoria_idade']=='Jovem Adulto'][phq['Gênero']==2]['Q05'].mean()
phq[phq['Categoria_idade']=='Jovem Adulto'][phq['Gênero']==2]['Q05'] = phq['Q05'].fillna(phq_05_media_pmja, inplace=True)
# Homem jovem adulto:
print('Média para homem jovem adulto:', phq[phq['Q05'].notnull()][phq['Categoria_idade']=='Jovem Adulto'][phq['Gênero']==1]['Q05'].mean())
phq_05_media_phja = phq[phq['Q05'].notnull()][phq['Categoria_idade']=='Jovem Adulto'][phq['Gênero']==1]['Q05'].mean()
phq[phq['Categoria_idade']=='Jovem Adulto'][phq['Gênero']==1]['Q05'] = phq['Q05'].fillna(phq_05_media_phja, inplace=True)
# Mulher adulta:
print('Média para mulher adulta:', phq[phq['Q05'].notnull()][phq['Categoria_idade']=='Adulto'][phq['Gênero']==2]['Q05'].mean())
phq_05_media_pma = phq[phq['Q05'].notnull()][phq['Categoria_idade']=='Adulto'][phq['Gênero']==2]['Q05'].mean()
phq[phq['Categoria_idade']=='Adulto'][phq['Gênero']==2]['Q05'] = phq['Q05'].fillna(phq_05_media_pma, inplace=True)
# Homem adulto:
print('Média para homem adulto:', phq[phq['Q05'].notnull()][phq['Categoria_idade']=='Adulto'][phq['Gênero']==1]['Q05'].mean())
phq_05_media_pha = phq[phq['Q05'].notnull()][phq['Categoria_idade']=='Adulto'][phq['Gênero']==1]['Q05'].mean()
phq[phq['Categoria_idade']=='Adulto'][phq['Gênero']==1]['Q05'] = phq['Q05'].fillna(phq_05_media_pha, inplace=True)

Gênero: 2    268
1    230
Name: Gênero, dtype: int64
Etnia: 3    180
4    152
1    111
5     37
2     18
Name: Etnia, dtype: int64
Categoria idade: Jovem Adulto    171
Adulto          155
Idoso           144
Adolescente      28
Bebê              0
Criança           0
Name: Categoria_idade, dtype: int64


C:\Users\Akihiro\AppData\Local\Temp/ipykernel_16380/962339829.py:7: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  print('Média para mulher jovem adulta:', phq[phq['Q05'].notnull()][phq['Categoria_idade']=='Jovem Adulto'][phq['Gênero']==2]['Q05'].mean())
C:\Users\Akihiro\AppData\Local\Temp/ipykernel_16380/962339829.py:8: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  phq_05_media_pmja = phq[phq['Q05'].notnull()][phq['Categoria_idade']=='Jovem Adulto'][phq['Gênero']==2]['Q05'].mean()
C:\Users\Akihiro\AppData\Local\Temp/ipykernel_16380/962339829.py:9: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  phq[phq['Categoria_idade']=='Jovem Adulto'][phq['Gênero']==2]['Q05'] = phq['Q05'].fillna(phq_05_media_pmja, inplace=True)
C:\Users\Akihiro\AppData\Local\Temp/ipykernel_16380/962339829.py:11: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  print('Média para homem jovem adult

Média para mulher jovem adulta: 0.418729817007535
Média para homem jovem adulto: 0.28109467266350163
Média para mulher adulta: 0.4407710954087879
Média para homem adulto: 0.26255977695036126


C:\Users\Akihiro\AppData\Local\Temp/ipykernel_16380/962339829.py:12: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  phq_05_media_phja = phq[phq['Q05'].notnull()][phq['Categoria_idade']=='Jovem Adulto'][phq['Gênero']==1]['Q05'].mean()
C:\Users\Akihiro\AppData\Local\Temp/ipykernel_16380/962339829.py:13: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  phq[phq['Categoria_idade']=='Jovem Adulto'][phq['Gênero']==1]['Q05'] = phq['Q05'].fillna(phq_05_media_phja, inplace=True)
C:\Users\Akihiro\AppData\Local\Temp/ipykernel_16380/962339829.py:15: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  print('Média para mulher adulta:', phq[phq['Q05'].notnull()][phq['Categoria_idade']=='Adulto'][phq['Gênero']==2]['Q05'].mean())
C:\Users\Akihiro\AppData\Local\Temp/ipykernel_16380/962339829.py:16: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  phq_05_media_pma = phq[phq['Q05'].notnull()]

In [14]:
# Q06
print('Gênero:', phq[phq['Q06'].isnull()]['Gênero'].value_counts())
print('Etnia:', phq[phq['Q06'].isnull()]['Etnia'].value_counts())
print('Categoria idade:', phq[phq['Q06'].isnull()]['Categoria_idade'].value_counts())
# Criar variável por gênero e idade:
# Mulher jovem adulta:
print('Média para mulher jovem adulta:', phq[phq['Q06'].notnull()][phq['Categoria_idade']=='Jovem Adulto'][phq['Gênero']==2]['Q06'].mean())
phq_06_media_pmja = phq[phq['Q06'].notnull()][phq['Categoria_idade']=='Jovem Adulto'][phq['Gênero']==2]['Q06'].mean()
phq[phq['Categoria_idade']=='Jovem Adulto'][phq['Gênero']==2]['Q06'] = phq['Q06'].fillna(phq_06_media_pmja, inplace=True)
# Homem jovem adulto:
print('Média para homem jovem adulto:', phq[phq['Q06'].notnull()][phq['Categoria_idade']=='Jovem Adulto'][phq['Gênero']==1]['Q06'].mean())
phq_06_media_phja = phq[phq['Q06'].notnull()][phq['Categoria_idade']=='Jovem Adulto'][phq['Gênero']==1]['Q06'].mean()
phq[phq['Categoria_idade']=='Jovem Adulto'][phq['Gênero']==1]['Q06'] = phq['Q06'].fillna(phq_06_media_phja, inplace=True)
# Mulher adulta:
print('Média para mulher adulta:', phq[phq['Q06'].notnull()][phq['Categoria_idade']=='Adulto'][phq['Gênero']==2]['Q06'].mean())
phq_06_media_pma = phq[phq['Q06'].notnull()][phq['Categoria_idade']=='Adulto'][phq['Gênero']==2]['Q06'].mean()
phq[phq['Categoria_idade']=='Adulto'][phq['Gênero']==2]['Q06'] = phq['Q06'].fillna(phq_06_media_pma, inplace=True)
# Homem adulto:
print('Média para homem adulto:', phq[phq['Q06'].notnull()][phq['Categoria_idade']=='Adulto'][phq['Gênero']==1]['Q06'].mean())
phq_06_media_pha = phq[phq['Q06'].notnull()][phq['Categoria_idade']=='Adulto'][phq['Gênero']==1]['Q06'].mean()
phq[phq['Categoria_idade']=='Adulto'][phq['Gênero']==1]['Q06'] = phq['Q06'].fillna(phq_06_media_pha, inplace=True)

Gênero: 2    268
1    230
Name: Gênero, dtype: int64
Etnia: 3    180
4    152
1    111
5     37
2     18
Name: Etnia, dtype: int64
Categoria idade: Jovem Adulto    171
Adulto          155
Idoso           144
Adolescente      28
Bebê              0
Criança           0
Name: Categoria_idade, dtype: int64
Média para mulher jovem adulta: 0.20882669537136705
Média para homem jovem adulto: 0.18636220770298428


C:\Users\Akihiro\AppData\Local\Temp/ipykernel_16380/3759569020.py:7: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  print('Média para mulher jovem adulta:', phq[phq['Q06'].notnull()][phq['Categoria_idade']=='Jovem Adulto'][phq['Gênero']==2]['Q06'].mean())
C:\Users\Akihiro\AppData\Local\Temp/ipykernel_16380/3759569020.py:8: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  phq_06_media_pmja = phq[phq['Q06'].notnull()][phq['Categoria_idade']=='Jovem Adulto'][phq['Gênero']==2]['Q06'].mean()
C:\Users\Akihiro\AppData\Local\Temp/ipykernel_16380/3759569020.py:9: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  phq[phq['Categoria_idade']=='Jovem Adulto'][phq['Gênero']==2]['Q06'] = phq['Q06'].fillna(phq_06_media_pmja, inplace=True)
C:\Users\Akihiro\AppData\Local\Temp/ipykernel_16380/3759569020.py:11: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  print('Média para homem jovem a

Média para mulher adulta: 0.30724805526838983
Média para homem adulto: 0.1891580934902678


C:\Users\Akihiro\AppData\Local\Temp/ipykernel_16380/3759569020.py:20: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  phq_06_media_pha = phq[phq['Q06'].notnull()][phq['Categoria_idade']=='Adulto'][phq['Gênero']==1]['Q06'].mean()
C:\Users\Akihiro\AppData\Local\Temp/ipykernel_16380/3759569020.py:21: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  phq[phq['Categoria_idade']=='Adulto'][phq['Gênero']==1]['Q06'] = phq['Q06'].fillna(phq_06_media_pha, inplace=True)


In [15]:
# Q07
print('Gênero:', phq[phq['Q07'].isnull()]['Gênero'].value_counts())
print('Etnia:', phq[phq['Q07'].isnull()]['Etnia'].value_counts())
print('Categoria idade:', phq[phq['Q07'].isnull()]['Categoria_idade'].value_counts())
# Criar variável por gênero e idade:
# Mulher jovem adulta:
print('Média para mulher jovem adulta:', phq[phq['Q07'].notnull()][phq['Categoria_idade']=='Jovem Adulto'][phq['Gênero']==2]['Q07'].mean())
phq_07_media_pmja = phq[phq['Q07'].notnull()][phq['Categoria_idade']=='Jovem Adulto'][phq['Gênero']==2]['Q07'].mean()
phq[phq['Categoria_idade']=='Jovem Adulto'][phq['Gênero']==2]['Q07'] = phq['Q07'].fillna(phq_07_media_pmja, inplace=True)
# Homem jovem adulto:
print('Média para homem jovem adulto:', phq[phq['Q07'].notnull()][phq['Categoria_idade']=='Jovem Adulto'][phq['Gênero']==1]['Q07'].mean())
phq_07_media_phja = phq[phq['Q07'].notnull()][phq['Categoria_idade']=='Jovem Adulto'][phq['Gênero']==1]['Q07'].mean()
phq[phq['Categoria_idade']=='Jovem Adulto'][phq['Gênero']==1]['Q07'] = phq['Q07'].fillna(phq_07_media_phja, inplace=True)
# Mulher adulta:
print('Média para mulher adulta:', phq[phq['Q07'].notnull()][phq['Categoria_idade']=='Adulto'][phq['Gênero']==2]['Q07'].mean())
phq_07_media_pma = phq[phq['Q07'].notnull()][phq['Categoria_idade']=='Adulto'][phq['Gênero']==2]['Q07'].mean()
phq[phq['Categoria_idade']=='Adulto'][phq['Gênero']==2]['Q07'] = phq['Q07'].fillna(phq_07_media_pma, inplace=True)
# Homem adulto:
print('Média para homem adulto:', phq[phq['Q07'].notnull()][phq['Categoria_idade']=='Adulto'][phq['Gênero']==1]['Q07'].mean())
phq_07_media_pha = phq[phq['Q07'].notnull()][phq['Categoria_idade']=='Adulto'][phq['Gênero']==1]['Q07'].mean()
phq[phq['Categoria_idade']=='Adulto'][phq['Gênero']==1]['Q07'] = phq['Q07'].fillna(phq_07_media_pha, inplace=True)

Gênero: 2    268
1    230
Name: Gênero, dtype: int64
Etnia: 3    180
4    152
1    111
5     37
2     18
Name: Etnia, dtype: int64
Categoria idade: Jovem Adulto    171
Adulto          155
Idoso           144
Adolescente      28
Bebê              0
Criança           0
Name: Categoria_idade, dtype: int64
Média para mulher jovem adulta: 0.22282023681377824
Média para homem jovem adulto: 0.21481379494496886
Média para mulher adulta: 0.3414521215307758


C:\Users\Akihiro\AppData\Local\Temp/ipykernel_16380/1601490916.py:7: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  print('Média para mulher jovem adulta:', phq[phq['Q07'].notnull()][phq['Categoria_idade']=='Jovem Adulto'][phq['Gênero']==2]['Q07'].mean())
C:\Users\Akihiro\AppData\Local\Temp/ipykernel_16380/1601490916.py:8: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  phq_07_media_pmja = phq[phq['Q07'].notnull()][phq['Categoria_idade']=='Jovem Adulto'][phq['Gênero']==2]['Q07'].mean()
C:\Users\Akihiro\AppData\Local\Temp/ipykernel_16380/1601490916.py:9: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  phq[phq['Categoria_idade']=='Jovem Adulto'][phq['Gênero']==2]['Q07'] = phq['Q07'].fillna(phq_07_media_pmja, inplace=True)
C:\Users\Akihiro\AppData\Local\Temp/ipykernel_16380/1601490916.py:11: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  print('Média para homem jovem a

Média para homem adulto: 0.2078220308575531


C:\Users\Akihiro\AppData\Local\Temp/ipykernel_16380/1601490916.py:21: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  phq[phq['Categoria_idade']=='Adulto'][phq['Gênero']==1]['Q07'] = phq['Q07'].fillna(phq_07_media_pha, inplace=True)


In [16]:
# Q08
print('Gênero:', phq[phq['Q08'].isnull()]['Gênero'].value_counts())
print('Etnia:', phq[phq['Q08'].isnull()]['Etnia'].value_counts())
print('Categoria idade:', phq[phq['Q08'].isnull()]['Categoria_idade'].value_counts())
# Criar variável por gênero e idade:
# Mulher jovem adulta:
print('Média para mulher jovem adulta:', phq[phq['Q08'].notnull()][phq['Categoria_idade']=='Jovem Adulto'][phq['Gênero']==2]['Q08'].mean())
phq_08_media_pmja = phq[phq['Q08'].notnull()][phq['Categoria_idade']=='Jovem Adulto'][phq['Gênero']==2]['Q08'].mean()
phq[phq['Categoria_idade']=='Jovem Adulto'][phq['Gênero']==2]['Q08'] = phq['Q08'].fillna(phq_08_media_pmja, inplace=True)
# Homem jovem adulto:
print('Média para homem jovem adulto:', phq[phq['Q08'].notnull()][phq['Categoria_idade']=='Jovem Adulto'][phq['Gênero']==1]['Q08'].mean())
phq_08_media_phja = phq[phq['Q08'].notnull()][phq['Categoria_idade']=='Jovem Adulto'][phq['Gênero']==1]['Q08'].mean()
phq[phq['Categoria_idade']=='Jovem Adulto'][phq['Gênero']==1]['Q08'] = phq['Q08'].fillna(phq_08_media_phja, inplace=True)
# Mulher adulta:
print('Média para mulher adulta:', phq[phq['Q08'].notnull()][phq['Categoria_idade']=='Adulto'][phq['Gênero']==2]['Q08'].mean())
phq_08_media_pma = phq[phq['Q08'].notnull()][phq['Categoria_idade']=='Adulto'][phq['Gênero']==2]['Q08'].mean()
phq[phq['Categoria_idade']=='Adulto'][phq['Gênero']==2]['Q08'] = phq['Q08'].fillna(phq_08_media_pma, inplace=True)
# Homem adulto:
print('Média para homem adulto:', phq[phq['Q08'].notnull()][phq['Categoria_idade']=='Adulto'][phq['Gênero']==1]['Q08'].mean())
phq_08_media_pha = phq[phq['Q08'].notnull()][phq['Categoria_idade']=='Adulto'][phq['Gênero']==1]['Q08'].mean()
phq[phq['Categoria_idade']=='Adulto'][phq['Gênero']==1]['Q08'] = phq['Q08'].fillna(phq_08_media_pha, inplace=True)

Gênero: 2    268
1    230
Name: Gênero, dtype: int64
Etnia: 3    180
4    152
1    111
5     37
2     18
Name: Etnia, dtype: int64
Categoria idade: Jovem Adulto    171
Adulto          155
Idoso           144
Adolescente      28
Bebê              0
Criança           0
Name: Categoria_idade, dtype: int64
Média para mulher jovem adulta: 0.12594187298170076
Média para homem jovem adulto: 0.13978521541041872
Média para mulher adulta: 0.20002581528335855
Média para homem adulto: 0.1373743434093889


C:\Users\Akihiro\AppData\Local\Temp/ipykernel_16380/1483705457.py:7: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  print('Média para mulher jovem adulta:', phq[phq['Q08'].notnull()][phq['Categoria_idade']=='Jovem Adulto'][phq['Gênero']==2]['Q08'].mean())
C:\Users\Akihiro\AppData\Local\Temp/ipykernel_16380/1483705457.py:8: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  phq_08_media_pmja = phq[phq['Q08'].notnull()][phq['Categoria_idade']=='Jovem Adulto'][phq['Gênero']==2]['Q08'].mean()
C:\Users\Akihiro\AppData\Local\Temp/ipykernel_16380/1483705457.py:9: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  phq[phq['Categoria_idade']=='Jovem Adulto'][phq['Gênero']==2]['Q08'] = phq['Q08'].fillna(phq_08_media_pmja, inplace=True)
C:\Users\Akihiro\AppData\Local\Temp/ipykernel_16380/1483705457.py:11: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  print('Média para homem jovem a

In [17]:
# Q09
print('Gênero:', phq[phq['Q09'].isnull()]['Gênero'].value_counts())
print('Etnia:', phq[phq['Q09'].isnull()]['Etnia'].value_counts())
print('Categoria idade:', phq[phq['Q09'].isnull()]['Categoria_idade'].value_counts())
# Criar variável por gênero e idade:
# Mulher jovem adulta:
print('Média para mulher jovem adulta:', phq[phq['Q09'].notnull()][phq['Categoria_idade']=='Jovem Adulto'][phq['Gênero']==2]['Q09'].mean())
phq_09_media_pmja = phq[phq['Q09'].notnull()][phq['Categoria_idade']=='Jovem Adulto'][phq['Gênero']==2]['Q09'].mean()
phq[phq['Categoria_idade']=='Jovem Adulto'][phq['Gênero']==2]['Q09'] = phq['Q09'].fillna(phq_09_media_pmja, inplace=True)
# Homem jovem adulto:
print('Média para homem jovem adulto:', phq[phq['Q09'].notnull()][phq['Categoria_idade']=='Jovem Adulto'][phq['Gênero']==1]['Q09'].mean())
phq_09_media_phja = phq[phq['Q09'].notnull()][phq['Categoria_idade']=='Jovem Adulto'][phq['Gênero']==1]['Q09'].mean()
phq[phq['Categoria_idade']=='Jovem Adulto'][phq['Gênero']==1]['Q09'] = phq['Q09'].fillna(phq_09_media_phja, inplace=True)
# Mulher adulta:
print('Média para mulher adulta:', phq[phq['Q09'].notnull()][phq['Categoria_idade']=='Adulto'][phq['Gênero']==2]['Q09'].mean())
phq_09_media_pma = phq[phq['Q09'].notnull()][phq['Categoria_idade']=='Adulto'][phq['Gênero']==2]['Q09'].mean()
phq[phq['Categoria_idade']=='Adulto'][phq['Gênero']==2]['Q09'] = phq['Q09'].fillna(phq_08_media_pma, inplace=True)
# Homem adulto:
print('Média para homem adulto:', phq[phq['Q09'].notnull()][phq['Categoria_idade']=='Adulto'][phq['Gênero']==1]['Q09'].mean())
phq_09_media_pha = phq[phq['Q09'].notnull()][phq['Categoria_idade']=='Adulto'][phq['Gênero']==1]['Q09'].mean()
phq[phq['Categoria_idade']=='Adulto'][phq['Gênero']==1]['Q09'] = phq['Q09'].fillna(phq_09_media_pha, inplace=True)

Gênero: 2    268
1    230
Name: Gênero, dtype: int64
Etnia: 3    180
4    152
1    111
5     37
2     18
Name: Etnia, dtype: int64
Categoria idade: Jovem Adulto    171
Adulto          155
Idoso           144
Adolescente      28
Bebê              0
Criança           0
Name: Categoria_idade, dtype: int64
Média para mulher jovem adulta: 0.03444564047362755
Média para homem jovem adulto: 0.05105236660797781
Média para mulher adulta: 0.08216365231786897
Média para homem adulto: 0.037871381766633155


C:\Users\Akihiro\AppData\Local\Temp/ipykernel_16380/1181558167.py:7: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  print('Média para mulher jovem adulta:', phq[phq['Q09'].notnull()][phq['Categoria_idade']=='Jovem Adulto'][phq['Gênero']==2]['Q09'].mean())
C:\Users\Akihiro\AppData\Local\Temp/ipykernel_16380/1181558167.py:8: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  phq_09_media_pmja = phq[phq['Q09'].notnull()][phq['Categoria_idade']=='Jovem Adulto'][phq['Gênero']==2]['Q09'].mean()
C:\Users\Akihiro\AppData\Local\Temp/ipykernel_16380/1181558167.py:9: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  phq[phq['Categoria_idade']=='Jovem Adulto'][phq['Gênero']==2]['Q09'] = phq['Q09'].fillna(phq_09_media_pmja, inplace=True)
C:\Users\Akihiro\AppData\Local\Temp/ipykernel_16380/1181558167.py:11: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  print('Média para homem jovem a

A pontuação das questões foi imputada com valores das médias por gênero e categoria_idade

In [18]:
phq.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5334 entries, 0 to 5333
Data columns (total 16 columns):
 #   Column           Non-Null Count  Dtype   
---  ------           --------------  -----   
 0   ID               5334 non-null   int64   
 1   Q01              4836 non-null   float64 
 2   Q02              5334 non-null   float64 
 3   Q03              5334 non-null   float64 
 4   Q04              5334 non-null   float64 
 5   Q05              5334 non-null   float64 
 6   Q06              5334 non-null   float64 
 7   Q07              5334 non-null   float64 
 8   Q08              5334 non-null   float64 
 9   Q09              5334 non-null   float64 
 10  Gênero           5334 non-null   int64   
 11  Idade            5334 non-null   int64   
 12  Etnia            5334 non-null   int64   
 13  Escolaridade     5334 non-null   int64   
 14  Renda_familiar   5287 non-null   float64 
 15  Categoria_idade  5334 non-null   category
dtypes: category(1), float64(10), int64(5)
memo

Grande parte dos valores nulos foi preenchida, falta apenas a variável renda_familiar para completar o tratamento de dados.
Por se tratar de poucos dados faltantes, irei aplicar apenas a média dos valores não nulos da coluna Renda_familiar e da Q01.

In [19]:
phq_renda_media = phq['Renda_familiar'].mean()
phq['Renda_familiar'].fillna(phq_renda_media, inplace=True)
phq_q01_media = phq['Q01'].mean()
phq['Q01'].fillna(phq_renda_media, inplace=True) 

In [20]:
phq.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5334 entries, 0 to 5333
Data columns (total 16 columns):
 #   Column           Non-Null Count  Dtype   
---  ------           --------------  -----   
 0   ID               5334 non-null   int64   
 1   Q01              5334 non-null   float64 
 2   Q02              5334 non-null   float64 
 3   Q03              5334 non-null   float64 
 4   Q04              5334 non-null   float64 
 5   Q05              5334 non-null   float64 
 6   Q06              5334 non-null   float64 
 7   Q07              5334 non-null   float64 
 8   Q08              5334 non-null   float64 
 9   Q09              5334 non-null   float64 
 10  Gênero           5334 non-null   int64   
 11  Idade            5334 non-null   int64   
 12  Etnia            5334 non-null   int64   
 13  Escolaridade     5334 non-null   int64   
 14  Renda_familiar   5334 non-null   float64 
 15  Categoria_idade  5334 non-null   category
dtypes: category(1), float64(10), int64(5)
memo

##### Banco

In [21]:
phq

ID       Q01       Q02       Q03       Q04      Q05       Q06  \
0     31130  8.738604  0.311087  0.622174  0.874058  0.41873  0.208827   
1     31131  0.000000  0.000000  0.000000  0.000000  0.00000  0.000000   
2     31132  0.000000  0.000000  0.000000  0.000000  0.00000  0.000000   
3     31134  0.000000  0.000000  0.000000  0.000000  0.00000  0.000000   
4     31139  0.000000  0.000000  0.000000  0.000000  3.00000  1.000000   
...     ...       ...       ...       ...       ...      ...       ...   
5329  41466  0.000000  0.000000  1.000000  1.000000  0.00000  0.000000   
5330  41468  0.000000  2.000000  0.000000  1.000000  1.00000  2.000000   
5331  41469  1.000000  0.000000  0.000000  1.000000  0.00000  0.000000   
5332  41472  0.000000  0.000000  0.000000  0.000000  0.00000  0.000000   
5333  41473  0.000000  0.000000  2.000000  1.000000  0.00000  1.000000   

          Q07       Q08       Q09  Gênero  Idade  Etnia  Escolaridade  \
0     0.22282  0.125942  0.034446       2     85      3             4   
1     0.00000  0.000000  0.000000       2     44      4             4   
2     0.00000  0.000000  0.000000       1     70      3             5   
3     0.00000  0.000000  0.000000       1     73      3             3   
4     0.00000  0.000000  0.000000       2     18      2             3   
...       ...       ...       ...     ...    ...    ...           ...   
5329  0.00000  0.000000  0.000000       2     58      5             2   
5330  1.00000  3.000000  0.000000       2     66      1             1   
5331  0.00000  0.000000  0.000000       1     19      4             4   
5332  0.00000  0.000000  0.000000       1     34      3             4   
5333  0.00000  0.000000  0.000000       1     21      1             2   

      Renda_familiar Categoria_idade  
0                4.0           Idoso  
1               11.0          Adulto  
2               11.0           Idoso  
3               12.0           Idoso  
4               11.0     Adolescente  
...              ...             ...  
5329             3.0          Adulto  
5330             8.0           Idoso  
5331             2.0    Jovem Adulto  
5332             7.0    Jovem Adulto  
5333             6.0    Jovem Adulto  

[5334 rows x 16 columns]

In [22]:
phq.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5334 entries, 0 to 5333
Data columns (total 16 columns):
 #   Column           Non-Null Count  Dtype   
---  ------           --------------  -----   
 0   ID               5334 non-null   int64   
 1   Q01              5334 non-null   float64 
 2   Q02              5334 non-null   float64 
 3   Q03              5334 non-null   float64 
 4   Q04              5334 non-null   float64 
 5   Q05              5334 non-null   float64 
 6   Q06              5334 non-null   float64 
 7   Q07              5334 non-null   float64 
 8   Q08              5334 non-null   float64 
 9   Q09              5334 non-null   float64 
 10  Gênero           5334 non-null   int64   
 11  Idade            5334 non-null   int64   
 12  Etnia            5334 non-null   int64   
 13  Escolaridade     5334 non-null   int64   
 14  Renda_familiar   5334 non-null   float64 
 15  Categoria_idade  5334 non-null   category
dtypes: category(1), float64(10), int64(5)
memo

### Preencher valores nulos do PAG_HEI

In [23]:
pag_hei.head()

ID     PAG_MINW  ADHERENCE  HEI2015C1_TOTALVEG  \
0  31128          NaN        NaN            3.083350   
1  31129   285.450346        2.0            1.697761   
2  31130     0.000000        1.0                 NaN   
3  31131   264.711316        2.0            3.115103   
4  31132  1353.348730        3.0            3.509081   

   HEI2015C2_GREEN_AND_BEAN  HEI2015C3_TOTALFRUIT  HEI2015C4_WHOLEFRUIT  \
0                       0.0              3.927008              5.000000   
1                       0.0              3.731602              4.929186   
2                       NaN                   NaN                   NaN   
3                       0.0              5.000000              1.320674   
4                       0.0              5.000000              5.000000   

   HEI2015C5_WHOLEGRAIN  HEI2015C6_TOTALDAIRY  HEI2015C7_TOTPROT  \
0              0.000000              5.000000           4.251329   
1              0.527631              3.887809           4.693141   
2                   NaN                   NaN                NaN   
3              0.000000              3.141062           4.071959   
4              5.000000              4.448512           5.000000   

   HEI2015C8_SEAPLANT_PROT  HEI2015C9_FATTYACID  HEI2015C10_SODIUM  \
0                 0.000000             2.487572           6.911882   
1                 0.928562             3.740758           6.146441   
2                      NaN                  NaN                NaN   
3                 2.052399             0.684985           2.297100   
4                 0.000000             2.050769           7.369489   

   HEI2015C11_REFINEDGRAIN  HEI2015C12_SFAT  HEI2015C13_ADDSUG  \
0                10.000000         0.818029           6.408092   
1                 7.274979         8.006457           0.000000   
2                      NaN              NaN                NaN   
3                 9.512964         0.438428           3.919226   
4                 8.887880         8.186877           9.099088   

   HEI2015_TOTAL_SCORE  
0            47.887262  
1            45.564326  
2                  NaN  
3            35.553899  
4            63.551696

In [24]:
pag_hei.tail()

ID    PAG_MINW  ADHERENCE  HEI2015C1_TOTALVEG  \
9419  41469  184.711316        2.0            2.555935   
9420  41471  951.478060        3.0            2.674192   
9421  41472   27.713626        1.0            2.457422   
9422  41473  685.912240        3.0            3.907340   
9423  41474    0.000000        1.0            1.238646   

      HEI2015C2_GREEN_AND_BEAN  HEI2015C3_TOTALFRUIT  HEI2015C4_WHOLEFRUIT  \
9419                  0.000000              2.613733              0.000000   
9420                  3.769401              2.198817              0.517369   
9421                  2.930799              2.311917              0.000000   
9422                  5.000000              0.000000              0.000000   
9423                  4.610515              0.688137              0.240848   

      HEI2015C5_WHOLEGRAIN  HEI2015C6_TOTALDAIRY  HEI2015C7_TOTPROT  \
9419              0.000000              2.151846           5.000000   
9420              0.463168              4.662005           5.000000   
9421              0.933303              5.000000           5.000000   
9422              0.000000              5.000000           5.000000   
9423              5.000000              5.000000           4.304982   

      HEI2015C8_SEAPLANT_PROT  HEI2015C9_FATTYACID  HEI2015C10_SODIUM  \
9419                 0.000000             9.344947           3.535211   
9420                 1.191796             6.145146           3.189620   
9421                 5.000000             2.090534           8.409844   
9422                 5.000000             0.000000           7.458058   
9423                 5.000000             2.162293           7.018993   

      HEI2015C11_REFINEDGRAIN  HEI2015C12_SFAT  HEI2015C13_ADDSUG  \
9419                 4.326307         5.269780           8.195564   
9420                 3.766149         3.974279           7.758940   
9421                 8.936854         3.063795           0.000000   
9422                10.000000         1.676912           2.751168   
9423                10.000000         7.541082           0.000000   

      HEI2015_TOTAL_SCORE  
9419            42.993323  
9420            45.310883  
9421            46.134467  
9422            45.793478  
9423            52.805494

In [25]:
pag_hei.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9424 entries, 0 to 9423
Data columns (total 17 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   ID                        9424 non-null   int64  
 1   PAG_MINW                  7150 non-null   float64
 2   ADHERENCE                 7150 non-null   float64
 3   HEI2015C1_TOTALVEG        8549 non-null   float64
 4   HEI2015C2_GREEN_AND_BEAN  8549 non-null   float64
 5   HEI2015C3_TOTALFRUIT      8549 non-null   float64
 6   HEI2015C4_WHOLEFRUIT      8549 non-null   float64
 7   HEI2015C5_WHOLEGRAIN      8549 non-null   float64
 8   HEI2015C6_TOTALDAIRY      8549 non-null   float64
 9   HEI2015C7_TOTPROT         8549 non-null   float64
 10  HEI2015C8_SEAPLANT_PROT   8549 non-null   float64
 11  HEI2015C9_FATTYACID       8549 non-null   float64
 12  HEI2015C10_SODIUM         8549 non-null   float64
 13  HEI2015C11_REFINEDGRAIN   8549 non-null   float64
 14  HEI2015C

In [26]:
pag_hei.isnull().sum()

ID                             0
PAG_MINW                    2274
ADHERENCE                   2274
HEI2015C1_TOTALVEG           875
HEI2015C2_GREEN_AND_BEAN     875
HEI2015C3_TOTALFRUIT         875
HEI2015C4_WHOLEFRUIT         875
HEI2015C5_WHOLEGRAIN         875
HEI2015C6_TOTALDAIRY         875
HEI2015C7_TOTPROT            875
HEI2015C8_SEAPLANT_PROT      875
HEI2015C9_FATTYACID          875
HEI2015C10_SODIUM            875
HEI2015C11_REFINEDGRAIN      875
HEI2015C12_SFAT              875
HEI2015C13_ADDSUG            875
HEI2015_TOTAL_SCORE          875
dtype: int64

In [27]:
print(pag_hei[pag_hei['PAG_MINW'].notnull()]['PAG_MINW'].mean())
print(pag_hei[pag_hei['ADHERENCE'].notnull()]['ADHERENCE'].mean())

565.1200668615468
2.074685314685315


In [28]:
lista_mean = []
for i in pag_hei:
    if pag_hei[i].isnull().sum()>0:
        lista_mean.append([i, pag_hei[pag_hei[i].notnull()][i].mean()])

In [29]:
lista_mean

[['PAG_MINW', 565.1200668615468],
 ['ADHERENCE', 2.074685314685315],
 ['HEI2015C1_TOTALVEG', 2.862097079587656],
 ['HEI2015C2_GREEN_AND_BEAN', 1.5478838387580773],
 ['HEI2015C3_TOTALFRUIT', 2.6599015054255504],
 ['HEI2015C4_WHOLEFRUIT', 2.40586399196248],
 ['HEI2015C5_WHOLEGRAIN', 1.6749491288864296],
 ['HEI2015C6_TOTALDAIRY', 4.054757182091902],
 ['HEI2015C7_TOTPROT', 4.211867021628733],
 ['HEI2015C8_SEAPLANT_PROT', 2.3253033773252416],
 ['HEI2015C9_FATTYACID', 4.2053196546959875],
 ['HEI2015C10_SODIUM', 4.988956364565648],
 ['HEI2015C11_REFINEDGRAIN', 5.540905877581909],
 ['HEI2015C12_SFAT', 5.781459465362386],
 ['HEI2015C13_ADDSUG', 5.959600373386709],
 ['HEI2015_TOTAL_SCORE', 48.21886486125872]]

In [30]:
pag_hei = pag_hei.copy()
for i, j in lista_mean:
    pag_hei[pag_hei[i].isnull()][i] = pag_hei[i].fillna(j, inplace=True)

In [31]:
pag_hei.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9424 entries, 0 to 9423
Data columns (total 17 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   ID                        9424 non-null   int64  
 1   PAG_MINW                  9424 non-null   float64
 2   ADHERENCE                 9424 non-null   float64
 3   HEI2015C1_TOTALVEG        9424 non-null   float64
 4   HEI2015C2_GREEN_AND_BEAN  9424 non-null   float64
 5   HEI2015C3_TOTALFRUIT      9424 non-null   float64
 6   HEI2015C4_WHOLEFRUIT      9424 non-null   float64
 7   HEI2015C5_WHOLEGRAIN      9424 non-null   float64
 8   HEI2015C6_TOTALDAIRY      9424 non-null   float64
 9   HEI2015C7_TOTPROT         9424 non-null   float64
 10  HEI2015C8_SEAPLANT_PROT   9424 non-null   float64
 11  HEI2015C9_FATTYACID       9424 non-null   float64
 12  HEI2015C10_SODIUM         9424 non-null   float64
 13  HEI2015C11_REFINEDGRAIN   9424 non-null   float64
 14  HEI2015C

In [32]:
pag_hei.nunique

<bound method DataFrame.nunique of          ID     PAG_MINW  ADHERENCE  HEI2015C1_TOTALVEG  \
0     31128   565.120067   2.074685            3.083350   
1     31129   285.450346   2.000000            1.697761   
2     31130     0.000000   1.000000            2.862097   
3     31131   264.711316   2.000000            3.115103   
4     31132  1353.348730   3.000000            3.509081   
...     ...          ...        ...                 ...   
9419  41469   184.711316   2.000000            2.555935   
9420  41471   951.478060   3.000000            2.674192   
9421  41472    27.713626   1.000000            2.457422   
9422  41473   685.912240   3.000000            3.907340   
9423  41474     0.000000   1.000000            1.238646   

      HEI2015C2_GREEN_AND_BEAN  HEI2015C3_TOTALFRUIT  HEI2015C4_WHOLEFRUIT  \
0                     0.000000              3.927008              5.000000   
1                     0.000000              3.731602              4.929186   
2                     

In [33]:
pag_hei

ID     PAG_MINW  ADHERENCE  HEI2015C1_TOTALVEG  \
0     31128   565.120067   2.074685            3.083350   
1     31129   285.450346   2.000000            1.697761   
2     31130     0.000000   1.000000            2.862097   
3     31131   264.711316   2.000000            3.115103   
4     31132  1353.348730   3.000000            3.509081   
...     ...          ...        ...                 ...   
9419  41469   184.711316   2.000000            2.555935   
9420  41471   951.478060   3.000000            2.674192   
9421  41472    27.713626   1.000000            2.457422   
9422  41473   685.912240   3.000000            3.907340   
9423  41474     0.000000   1.000000            1.238646   

      HEI2015C2_GREEN_AND_BEAN  HEI2015C3_TOTALFRUIT  HEI2015C4_WHOLEFRUIT  \
0                     0.000000              3.927008              5.000000   
1                     0.000000              3.731602              4.929186   
2                     1.547884              2.659902              2.405864   
3                     0.000000              5.000000              1.320674   
4                     0.000000              5.000000              5.000000   
...                        ...                   ...                   ...   
9419                  0.000000              2.613733              0.000000   
9420                  3.769401              2.198817              0.517369   
9421                  2.930799              2.311917              0.000000   
9422                  5.000000              0.000000              0.000000   
9423                  4.610515              0.688137              0.240848   

      HEI2015C5_WHOLEGRAIN  HEI2015C6_TOTALDAIRY  HEI2015C7_TOTPROT  \
0                 0.000000              5.000000           4.251329   
1                 0.527631              3.887809           4.693141   
2                 1.674949              4.054757           4.211867   
3                 0.000000              3.141062           4.071959   
4                 5.000000              4.448512           5.000000   
...                    ...                   ...                ...   
9419              0.000000              2.151846           5.000000   
9420              0.463168              4.662005           5.000000   
9421              0.933303              5.000000           5.000000   
9422              0.000000              5.000000           5.000000   
9423              5.000000              5.000000           4.304982   

      HEI2015C8_SEAPLANT_PROT  HEI2015C9_FATTYACID  HEI2015C10_SODIUM  \
0                    0.000000             2.487572           6.911882   
1                    0.928562             3.740758           6.146441   
2                    2.325303             4.205320           4.988956   
3                    2.052399             0.684985           2.297100   
4                    0.000000             2.050769           7.369489   
...                       ...                  ...                ...   
9419                 0.000000             9.344947           3.535211   
9420                 1.191796             6.145146           3.189620   
9421                 5.000000             2.090534           8.409844   
9422                 5.000000             0.000000           7.458058   
9423                 5.000000             2.162293           7.018993   

      HEI2015C11_REFINEDGRAIN  HEI2015C12_SFAT  HEI2015C13_ADDSUG  \
0                   10.000000         0.818029           6.408092   
1                    7.274979         8.006457           0.000000   
2                    5.540906         5.781459           5.959600   
3                    9.512964         0.438428           3.919226   
4                    8.887880         8.186877           9.099088   
...                       ...              ...                ...   
9419                 4.326307         5.269780           8.195564   
9420                 3.766149         3.974279           7.758940   
9421        

### Unir phq e pag_hei
Chave única é ID

In [34]:
phq_full = phq.merge(pag_hei, how='left', on='ID')

### Tratamento de dados

In [35]:
phq_full['phq9'] = phq_full['Q01'] + phq_full['Q02'] + phq_full['Q03'] + phq_full['Q04'] + phq_full['Q05'] + phq_full['Q06'] + phq_full['Q07'] + phq_full['Q08'] + phq_full['Q09']  

In [36]:
phq_full.head()

ID       Q01       Q02       Q03       Q04      Q05       Q06      Q07  \
0  31130  8.738604  0.311087  0.622174  0.874058  0.41873  0.208827  0.22282   
1  31131  0.000000  0.000000  0.000000  0.000000  0.00000  0.000000  0.00000   
2  31132  0.000000  0.000000  0.000000  0.000000  0.00000  0.000000  0.00000   
3  31134  0.000000  0.000000  0.000000  0.000000  0.00000  0.000000  0.00000   
4  31139  0.000000  0.000000  0.000000  0.000000  3.00000  1.000000  0.00000   

        Q08       Q09  ...  HEI2015C6_TOTALDAIRY  HEI2015C7_TOTPROT  \
0  0.125942  0.034446  ...              4.054757           4.211867   
1  0.000000  0.000000  ...              3.141062           4.071959   
2  0.000000  0.000000  ...              4.448512           5.000000   
3  0.000000  0.000000  ...              2.313900           5.000000   
4  0.000000  0.000000  ...              0.000000           4.088849   

   HEI2015C8_SEAPLANT_PROT  HEI2015C9_FATTYACID  HEI2015C10_SODIUM  \
0                 2.325303             4.205320           4.988956   
1                 2.052399             0.684985           2.297100   
2                 0.000000             2.050769           7.369489   
3                 1.268648             2.958985           0.000000   
4                 4.221442            10.000000           2.845774   

  HEI2015C11_REFINEDGRAIN  HEI2015C12_SFAT  HEI2015C13_ADDSUG  \
0                5.540906         5.781459           5.959600   
1                9.512964         0.438428           3.919226   
2                8.887880         8.186877           9.099088   
3                5.627488         5.863078           5.878686   
4                0.000000        10.000000           9.658507   

   HEI2015_TOTAL_SCORE       phq9  
0            48.218865  11.556688  
1            35.553899   0.000000  
2            63.551696   0.000000  
3            42.886826   0.000000  
4            47.074393   4.000000  

[5 rows x 33 columns]

In [37]:
phq_full.describe()

ID          Q01          Q02          Q03          Q04  \
count   5334.000000  5334.000000  5334.000000  5334.000000  5334.000000   
mean   36358.796775     1.094455     0.312884     0.557901     0.731961   
std     2964.731827     2.557336     0.676688     0.886864     0.871199   
min    31130.000000     0.000000     0.000000     0.000000     0.000000   
25%    33838.500000     0.000000     0.000000     0.000000     0.000000   
50%    36347.500000     0.000000     0.000000     0.000000     0.874058   
75%    38915.750000     1.000000     0.311087     1.000000     1.000000   
max    41473.000000     9.000000     9.000000     9.000000     9.000000   

               Q05          Q06          Q07          Q08          Q09  ...  \
count  5334.000000  5334.000000  5334.000000  5334.000000  5334.000000  ...   
mean      0.329870     0.209785     0.229652     0.145804     0.048773  ...   
std       0.708087     0.594677     0.608615     0.562185     0.311328  ...   
min       0.000000     0.000000     0.000000     0.000000     0.000000  ...   
25%       0.000000     0.000000     0.000000     0.000000     0.000000  ...   
50%       0.000000     0.000000     0.000000     0.000000     0.000000  ...   
75%       0.418730     0.000000     0.000000     0.000000     0.000000  ...   
max       9.000000     9.000000     9.000000     9.000000     9.000000  ...   

       HEI2015C6_TOTALDAIRY  HEI2015C7_TOTPROT  HEI2015C8_SEAPLANT_PROT  \
count           5334.000000        5334.000000              5334.000000   
mean               3.772265           4.422814                 2.579610   
std                1.564827           0.983294                 2.116136   
min                0.000000           0.000000                 0.000000   
25%                2.740337           4.211867                 0.043092   
50%                4.676100           5.000000                 2.325303   
75%                5.000000           5.000000                 5.000000   
max                5.000000           5.000000                 5.000000   

       HEI2015C9_FATTYACID  HEI2015C10_SODIUM  HEI2015C11_REFINEDGRAIN  \
count          5334.000000        5334.000000              5334.000000   
mean              4.681216           4.773906                 5.785707   
std               3.154750           3.331606                 3.405738   
min               0.000000           0.000000                 0.000000   
25%               2.163329           1.784006                 3.206577   
50%               4.211556           4.988956                 5.937633   
75%               7.061821           7.482189                 9.060863   
max              10.000000          10.000000                10.000000   

       HEI2015C12_SFAT  HEI2015C13_ADDSUG  HEI2015_TOTAL_SCORE         phq9  
count      5334.000000        5334.000000          5334.000000  5334.000000  
mean          6.053267           6.206971            49.881395     3.661086  
std           3.096110           3.244135            12.009986     4.589136  
min           0.000000           0.000000            16.059212     0.000000  
25%           3.861184           3.921979            41.504247     0.000000  
50%           6.161413           6.561476            48.605034     2.000000  
75%           8.851163           9.270889            57.594265     5.000000  
max          10.000000          10.000000            89.529383    42.000000  

[8 rows x 32 columns]

In [38]:
phq_full['phq_grp'] = phq_full['phq9']
for i in range(len(phq_full['phq9'])):
    if phq_full['phq9'][i]<5.1:
        phq_full['phq_grp'][i] = 'sem sintomas'
    elif 5<=phq_full['phq9'][i]<10.2:
        phq_full['phq_grp'][i] = 'sintomas leves'
    elif 10<=phq_full['phq9'][i]<15.3:
        phq_full['phq_grp'][i] = 'sintomas moderados'
    elif 15<=phq_full['phq9'][i]<=19:
        phq_full['phq_grp'][i] = 'sintomas moderadamente severos'
    elif phq_full['phq9'][i]>=20:
        phq_full['phq_grp'][i] = 'sintomas severos'        


C:\Users\Akihiro\AppData\Local\Temp/ipykernel_16380/3198925661.py:8: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  phq_full['phq_grp'][i] = 'sintomas moderados'
C:\Users\Akihiro\anaconda3\lib\site-packages\pandas\core\indexing.py:1732: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self._setitem_single_block(indexer, value, name)


In [39]:
phq_full['phq_grp'].value_counts()

sem sintomas                      4013
sintomas moderados                 673
sintomas leves                     555
sintomas moderadamente severos      58
sintomas severos                    35
Name: phq_grp, dtype: int64

In [40]:
phq_full['phq_grp']

0       sintomas moderados
1             sem sintomas
2             sem sintomas
3             sem sintomas
4             sem sintomas
               ...        
5329          sem sintomas
5330        sintomas leves
5331          sem sintomas
5332          sem sintomas
5333          sem sintomas
Name: phq_grp, Length: 5334, dtype: object

### 3- Análise Univariada

In [41]:
import seaborn as sns
import matplotlib.pyplot as plt

In [42]:
phq_full.describe()

ID          Q01          Q02          Q03          Q04  \
count   5334.000000  5334.000000  5334.000000  5334.000000  5334.000000   
mean   36358.796775     1.094455     0.312884     0.557901     0.731961   
std     2964.731827     2.557336     0.676688     0.886864     0.871199   
min    31130.000000     0.000000     0.000000     0.000000     0.000000   
25%    33838.500000     0.000000     0.000000     0.000000     0.000000   
50%    36347.500000     0.000000     0.000000     0.000000     0.874058   
75%    38915.750000     1.000000     0.311087     1.000000     1.000000   
max    41473.000000     9.000000     9.000000     9.000000     9.000000   

               Q05          Q06          Q07          Q08          Q09  ...  \
count  5334.000000  5334.000000  5334.000000  5334.000000  5334.000000  ...   
mean      0.329870     0.209785     0.229652     0.145804     0.048773  ...   
std       0.708087     0.594677     0.608615     0.562185     0.311328  ...   
min       0.000000     0.000000     0.000000     0.000000     0.000000  ...   
25%       0.000000     0.000000     0.000000     0.000000     0.000000  ...   
50%       0.000000     0.000000     0.000000     0.000000     0.000000  ...   
75%       0.418730     0.000000     0.000000     0.000000     0.000000  ...   
max       9.000000     9.000000     9.000000     9.000000     9.000000  ...   

       HEI2015C6_TOTALDAIRY  HEI2015C7_TOTPROT  HEI2015C8_SEAPLANT_PROT  \
count           5334.000000        5334.000000              5334.000000   
mean               3.772265           4.422814                 2.579610   
std                1.564827           0.983294                 2.116136   
min                0.000000           0.000000                 0.000000   
25%                2.740337           4.211867                 0.043092   
50%                4.676100           5.000000                 2.325303   
75%                5.000000           5.000000                 5.000000   
max                5.000000           5.000000                 5.000000   

       HEI2015C9_FATTYACID  HEI2015C10_SODIUM  HEI2015C11_REFINEDGRAIN  \
count          5334.000000        5334.000000              5334.000000   
mean              4.681216           4.773906                 5.785707   
std               3.154750           3.331606                 3.405738   
min               0.000000           0.000000                 0.000000   
25%               2.163329           1.784006                 3.206577   
50%               4.211556           4.988956                 5.937633   
75%               7.061821           7.482189                 9.060863   
max              10.000000          10.000000                10.000000   

       HEI2015C12_SFAT  HEI2015C13_ADDSUG  HEI2015_TOTAL_SCORE         phq9  
count      5334.000000        5334.000000          5334.000000  5334.000000  
mean          6.053267           6.206971            49.881395     3.661086  
std           3.096110           3.244135            12.009986     4.589136  
min           0.000000           0.000000            16.059212     0.000000  
25%           3.861184           3.921979            41.504247     0.000000  
50%           6.161413           6.561476            48.605034     2.000000  
75%           8.851163           9.270889            57.594265     5.000000  
max          10.000000          10.000000            89.529383    42.000000  

[8 rows x 32 columns]

In [43]:
phq_full.columns

Index(['ID', 'Q01', 'Q02', 'Q03', 'Q04', 'Q05', 'Q06', 'Q07', 'Q08', 'Q09',
       'Gênero', 'Idade', 'Etnia', 'Escolaridade', 'Renda_familiar',
       'Categoria_idade', 'PAG_MINW', 'ADHERENCE', 'HEI2015C1_TOTALVEG',
       'HEI2015C2_GREEN_AND_BEAN', 'HEI2015C3_TOTALFRUIT',
       'HEI2015C4_WHOLEFRUIT', 'HEI2015C5_WHOLEGRAIN', 'HEI2015C6_TOTALDAIRY',
       'HEI2015C7_TOTPROT', 'HEI2015C8_SEAPLANT_PROT', 'HEI2015C9_FATTYACID',
       'HEI2015C10_SODIUM', 'HEI2015C11_REFINEDGRAIN', 'HEI2015C12_SFAT',
       'HEI2015C13_ADDSUG', 'HEI2015_TOTAL_SCORE', 'phq9', 'phq_grp'],
      dtype='object')

In [44]:
age_cat_edges = [0, 3, 12, 18, 35, 60, 1000]
phq_full['Cat_idade_int'] = pd.cut(phq['Idade'], 
                                age_cat_edges, 
                                labels=[1, 2, 3, 4, 5, 6])

In [45]:
phq_full.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 5334 entries, 0 to 5333
Data columns (total 35 columns):
 #   Column                    Non-Null Count  Dtype   
---  ------                    --------------  -----   
 0   ID                        5334 non-null   int64   
 1   Q01                       5334 non-null   float64 
 2   Q02                       5334 non-null   float64 
 3   Q03                       5334 non-null   float64 
 4   Q04                       5334 non-null   float64 
 5   Q05                       5334 non-null   float64 
 6   Q06                       5334 non-null   float64 
 7   Q07                       5334 non-null   float64 
 8   Q08                       5334 non-null   float64 
 9   Q09                       5334 non-null   float64 
 10  Gênero                    5334 non-null   int64   
 11  Idade                     5334 non-null   int64   
 12  Etnia                     5334 non-null   int64   
 13  Escolaridade              5334 non-null   int64 

In [46]:
phq_full[['Cat_idade_int']] = phq_full[['Cat_idade_int']].apply(pd.to_numeric)

In [47]:
phq_full[['Gênero', 'Cat_idade_int', 'Etnia', 'Escolaridade', 'Renda_familiar', 'phq9']].describe().round(2)

Gênero  Cat_idade_int    Etnia  Escolaridade  Renda_familiar     phq9
count  5334.00        5334.00  5334.00       5334.00         5334.00  5334.00
mean      1.52           4.81     2.87          3.23            8.74     3.66
std       0.50           0.89     1.13          1.27           12.72     4.59
min       1.00           3.00     1.00          1.00            1.00     0.00
25%       1.00           4.00     3.00          2.00            5.00     0.00
50%       2.00           5.00     3.00          3.00            7.00     2.00
75%       2.00           6.00     4.00          4.00           10.00     5.00
max       2.00           6.00     5.00          9.00           99.00    42.00

In [48]:
print(phq_full[['Gênero']].value_counts())
print(phq_full[['Cat_idade_int']].value_counts())
print(phq_full[['Etnia']].value_counts())
print(phq_full[['Escolaridade']].value_counts())
print(phq_full[['Renda_familiar']].value_counts())
print(phq_full[['phq_grp']].value_counts())

Gênero
2         2773
1         2561
dtype: int64
Cat_idade_int
5                1824
4                1801
6                1408
3                 301
dtype: int64
Etnia
3        2516
4        1300
1        1133
5         221
2         164
dtype: int64
Escolaridade
4               1498
3               1335
5                939
2                936
1                618
7                  4
9                  4
dtype: int64
Renda_familiar
11.000000         1044
6.000000           678
7.000000           535
8.000000           485
3.000000           440
5.000000           426
4.000000           412
9.000000           297
2.000000           281
10.000000          278
1.000000           190
12.000000           78
99.000000           77
8.738604            47
77.000000           36
13.000000           30
dtype: int64
phq_grp                       
sem sintomas                      4013
sintomas moderados                 673
sintomas leves                     555
sintomas moderadamente severo

#### Conclusão da análise univariada
* Gênero: mediana de 2 e média de 1.52 indicam que existem mais mulheres no dataset do que homens.
* Cat_idade_int: 50% da amostra é composta por jovens adultos, adultos e idosos.
* Etnia: 50% dos valores estão entre 3 e 4, entre mexicano-americano e outras etnias.
* Escolaridade: 50% da amostra é composta por escolaridade do 9-12 ano até ensino superior incompleto.
* Renda familiar: 50% da amostra é composta por renda familiar entre 20.000 e 74.999 dólares anuais. E a mediana é a faixa de 35000-44999 dólares anuais.

### 4 - Análise bivariada
1. Faça uma análise bivariada de sintomas de depressão (phq9 ou phq_grp) com as características demográficas. Descreva o perfil com maiores incidências de sintomas de depressão.

In [49]:
phq_full[['Gênero', 'Cat_idade_int', 'Etnia', 'Escolaridade', 'Renda_familiar', 'phq_grp']].groupby('phq_grp').agg(['mean', 'median'])

Gênero        Cat_idade_int         \
                                    mean median          mean median   
phq_grp                                                                
sem sintomas                    1.501121    2.0      4.810117    5.0   
sintomas leves                  1.603604    2.0      4.776577    5.0   
sintomas moderadamente severos  1.568966    2.0      4.913793    5.0   
sintomas moderados              1.549777    2.0      4.842496    5.0   
sintomas severos                1.685714    2.0      5.057143    5.0   

                                   Etnia        Escolaridade         \
                                    mean median         mean median   
phq_grp                                                               
sem sintomas                    2.857214    3.0     3.309993    3.0   
sintomas leves                  2.859459    3.0     3.068468    3.0   
sintomas moderadamente severos  2.706897    3.0     3.017241    3.0   
sintomas moderados              2.970282    3.0     2.970282    3.0   
sintomas severos                3.000000    3.0     2.457143    2.0   

                               Renda_familiar         
                                         mean median  
phq_grp                                               
sem sintomas                         8.507218    7.0  
sintomas leves                       8.817383    6.0  
sintomas moderadamente severos       4.913793    5.0  
sintomas moderados                  10.351563    6.0  
sintomas severos                     9.342857    4.0

* O perfil das pessoas que apresentaram sintomas severos de depressão:
* Mulher adulta de etnia mexicano-americana com escolaridade entre o 9 até o 12 ano e renda familiar de 15000-19999 de dólares anuais.
* Esse perfil de sintomas severos é semelhante ao perfil de pessoas sem sintomas, há diferença na mediana da escolaridade e da renda familiar.
* A escolaridade e renda familiar são variáveis que contribuem com os sintomas de depressão?

2. Faça uma análise bivariada de características demográficas x hábitos saudáveis. Qual perfil possui hábitos mais saudáveis?

Irei criar uma categoria a partir do score total HEI para ter perfis com mais de 30 pessoas.

In [50]:
score_cut_edges = [0, 20, 40, 60, 80, 100]
phq_full['CAT_TOTAL_SCORE'] = pd.cut(phq_full['HEI2015_TOTAL_SCORE'], score_cut_edges, labels=['Severamente pouco saudável', 'Pouco saudável', 'Moderadamente saudável', 'Muito saudável', 'Extremamente saudável'])

In [51]:
phq_full[['Gênero', 'Cat_idade_int', 'Etnia', 'Escolaridade', 'Renda_familiar', 'phq_grp', 'CAT_TOTAL_SCORE', 'PAG_MINW']].groupby('CAT_TOTAL_SCORE').agg(['mean', 'median', 'count']).sort_values(by='CAT_TOTAL_SCORE', ascending=False)

Gênero              Cat_idade_int               \
                                mean median count          mean median count   
CAT_TOTAL_SCORE                                                                
Extremamente saudável       1.680000    2.0    50      5.380000    5.5    50   
Muito saudável              1.579882    2.0  1014      5.184418    5.0  1014   
Moderadamente saudável      1.511621    2.0  3141      4.811843    5.0  3141   
Pouco saudável              1.480322    1.0  1118      4.457961    4.0  1118   
Severamente pouco saudável  1.636364    2.0    11      4.636364    4.0    11   

                               Etnia              Escolaridade               \
                                mean median count         mean median count   
CAT_TOTAL_SCORE                                                               
Extremamente saudável       3.020000    3.0    50     3.780000    4.0    50   
Muito saudável              2.788955    3.0  1014     3.424063    4.0  1014   
Moderadamente saudável      2.856415    3.0  3141     3.196116    3.0  3141   
Pouco saudável              2.979428    3.0  1118     3.140429    3.0  1118   
Severamente pouco saudável  2.909091    3.0    11     3.181818    4.0    11   

                           Renda_familiar                 PAG_MINW  \
                                     mean median count        mean   
CAT_TOTAL_SCORE                                                      
Extremamente saudável            9.900000   10.0    50  651.967206   
Muito saudável                   9.431863    8.0  1014  482.883247   
Moderadamente saudável           8.719650    7.0  3141  455.187913   
Pouco saudável                   8.119308    7.0  1118  496.268396   
Severamente pouco saudável       7.909091    7.0    11  871.194625   

                                              
                                median count  
CAT_TOTAL_SCORE                               
Extremamente saudável       522.517321    50  
Muito saudável              238.452656  1014  
Moderadamente saudável      198.614319  3141  
Pouco saudável              215.854503  1118  
Severamente pouco saudável   69.284065    11

* **Perfil extremamente saudável**: Mulher adulta de etnia mexicano-americana com ensino superior incompleto e renda familiar de  65000-74999 de dólares anuais.

3. Faça uma análise bivariada de sintomas de depressão (phq9 ou phq_grp) com os hábitos saudáveis (variáveis de atividade física e healthy eating index). O que podemos observar?

In [52]:
phq_full.columns

Index(['ID', 'Q01', 'Q02', 'Q03', 'Q04', 'Q05', 'Q06', 'Q07', 'Q08', 'Q09',
       'Gênero', 'Idade', 'Etnia', 'Escolaridade', 'Renda_familiar',
       'Categoria_idade', 'PAG_MINW', 'ADHERENCE', 'HEI2015C1_TOTALVEG',
       'HEI2015C2_GREEN_AND_BEAN', 'HEI2015C3_TOTALFRUIT',
       'HEI2015C4_WHOLEFRUIT', 'HEI2015C5_WHOLEGRAIN', 'HEI2015C6_TOTALDAIRY',
       'HEI2015C7_TOTPROT', 'HEI2015C8_SEAPLANT_PROT', 'HEI2015C9_FATTYACID',
       'HEI2015C10_SODIUM', 'HEI2015C11_REFINEDGRAIN', 'HEI2015C12_SFAT',
       'HEI2015C13_ADDSUG', 'HEI2015_TOTAL_SCORE', 'phq9', 'phq_grp',
       'Cat_idade_int', 'CAT_TOTAL_SCORE'],
      dtype='object')

In [53]:
phq_full[['phq_grp', 'HEI2015_TOTAL_SCORE', 'PAG_MINW']].groupby('phq_grp').agg(['mean', 'median', 'count'])

HEI2015_TOTAL_SCORE                   \
                                              mean     median count   
phq_grp                                                               
sem sintomas                             50.137804  49.510152  4013   
sintomas leves                           48.998695  48.218865   555   
sintomas moderadamente severos           47.483475  46.464736    58   
sintomas moderados                       49.371952  48.218865   673   
sintomas severos                         48.248895  48.443689    35   

                                  PAG_MINW                    
                                      mean      median count  
phq_grp                                                       
sem sintomas                    500.895801  235.565820  4013  
sintomas leves                  451.418825  180.138568   555  
sintomas moderadamente severos  231.771920   89.491917    58  
sintomas moderados              342.004362  110.854503   673  
sintomas severos                347.268954    2.309469    35

* O HEI não está ligado aos sintomas de depressão
* O tempo de exercício físico por semana está ligado de certa forma aos sintomas de depressão. As pessoas que não apresentam sintomas de depressão se exercitam, em mediana, 235 min/semana. Nota-se uma diminuição de min de exercício/semana conforme os sintomas de depressão se intensificam.

In [54]:
phq_full[['Gênero', 'Cat_idade_int', 'Etnia', 'Escolaridade', 'Renda_familiar', 'phq_grp', 'CAT_TOTAL_SCORE', 'PAG_MINW']].groupby('phq_grp').agg(['mean', 'median', 'count', 'std']).sort_values(by='phq_grp', ascending=False)

Gênero                        Cat_idade_int  \
                                    mean median count       std          mean   
phq_grp                                                                         
sintomas severos                1.685714    2.0    35  0.471008      5.057143   
sintomas moderados              1.549777    2.0   673  0.497886      4.842496   
sintomas moderadamente severos  1.568966    2.0    58  0.499546      4.913793   
sintomas leves                  1.603604    2.0   555  0.489590      4.776577   
sem sintomas                    1.501121    2.0  4013  0.500061      4.810117   

                                                          Etnia         ...  \
                               median count       std      mean median  ...   
phq_grp                                                                 ...   
sintomas severos                  5.0    35  0.725293  3.000000    3.0  ...   
sintomas moderados                5.0   673  0.889175  2.970282    3.0  ...   
sintomas moderadamente severos    5.0    58  0.732323  2.706897    3.0  ...   
sintomas leves                    5.0   555  0.866854  2.859459    3.0  ...   
sem sintomas                      5.0  4013  0.896891  2.857214    3.0  ...   

                               Escolaridade           Renda_familiar         \
                                      count       std           mean median   
phq_grp                                                                       
sintomas severos                         35  1.146423       9.342857    4.0   
sintomas moderados                      673  1.338201      10.351563    6.0   
sintomas moderadamente severos           58  1.249440       4.913793    5.0   
sintomas leves                          555  1.216165       8.817383    6.0   
sem sintomas                           4013  1.254540       8.507218    7.0   

                                                   PAG_MINW                    \
                               count        std        mean      median count   
phq_grp                                                                         
sintomas severos                  35  20.011677  347.268954    2.309469    35   
sintomas moderados               673  18.466689  342.004362  110.854503   673   
sintomas moderadamente severos    58   2.657558  231.771920   89.491917    58   
sintomas leves                   555  14.923336  451.418825  180.138568   555   
sem sintomas                    4013  11.092584  500.895801  235.565820  4013   

                                             
                                        std  
phq_grp                                      
sintomas severos                1016.089064  
sintomas moderados               596.700622  
sintomas moderadamente severos   559.382630  
sintomas leves                   846.764921  
sem sintomas                     795.559346  

[5 rows x 24 columns]

### 5 - Testes de hipóteses

1. Faça o(s) teste(s) de hipóteses adequado(s) para avaliar a significância estatística das diferenças nas características demográficas apontadas na análise bivariada. Quais são as características apresentam diferenças estatisticamente significativas para a frequência de sintomas de depressão?
* Renda familiar: stat=-0.440, p=0.660
    * Hipótese nula
* Gênero: stat=-2.175, p=0.030
    * Hipótese alternativa
* Idade: stat=-1.625, p=0.104
    * Hipótese nula
* Etnia: stat=-0.758, p=0.449
    * Hipótese nula
* Escolaridade: stat=4.007, p=0.000
    * Hipótese alternativa
* Gênero e escolaridade são características estatisticamente significativas de acordo com o p-valor para sintomas de depressão.

Teste-t independente

* Hipótese nula: média de renda familiar do grupo sem sintomas == média de renda familiar do grupo com sintomas severos
* Hipótese alternativa: média de renda familiar do grupo sem sintomas != média de renda familiar do grupo com sintomas severos 

In [55]:
from scipy.stats import ttest_ind
stat, p = ttest_ind(phq_full[phq_full['phq_grp']=='sem sintomas']['Renda_familiar'], phq_full[phq_full['phq_grp']=='sintomas severos']['Renda_familiar'])
print('Renda familiar: stat=%.3f, p=%.3f' % (stat, p))
if p>0.05:
    print('Hipótese nula')
else:
     print('Hipótese alternativa')
stat, p = ttest_ind(phq_full[phq_full['phq_grp']=='sem sintomas']['Gênero'], phq_full[phq_full['phq_grp']=='sintomas severos']['Gênero'])
print('Gênero: stat=%.3f, p=%.3f' % (stat, p))
if p>0.05:
    print('Hipótese nula')
else:
     print('Hipótese alternativa')
stat, p = ttest_ind(phq_full[phq_full['phq_grp']=='sem sintomas']['Cat_idade_int'], phq_full[phq_full['phq_grp']=='sintomas severos']['Cat_idade_int'])
print('Idade: stat=%.3f, p=%.3f' % (stat, p))
if p>0.05:
    print('Hipótese nula')
else:
     print('Hipótese alternativa')

stat, p = ttest_ind(phq_full[phq_full['phq_grp']=='sem sintomas']['Etnia'], phq_full[phq_full['phq_grp']=='sintomas severos']['Etnia'])
print('Etnia: stat=%.3f, p=%.3f' % (stat, p))
if p>0.05:
    print('Hipótese nula')
else:
     print('Hipótese alternativa')
        
stat, p = ttest_ind(phq_full[phq_full['phq_grp']=='sem sintomas']['Escolaridade'], phq_full[phq_full['phq_grp']=='sintomas severos']['Escolaridade'])
print('Escolaridade: stat=%.3f, p=%.3f' % (stat, p))
if p>0.05:
    print('Hipótese nula')
else:
     print('Hipótese alternativa')


Renda familiar: stat=-0.440, p=0.660
Hipótese nula
Gênero: stat=-2.175, p=0.030
Hipótese alternativa
Idade: stat=-1.625, p=0.104
Hipótese nula
Etnia: stat=-0.758, p=0.449
Hipótese nula
Escolaridade: stat=4.007, p=0.000
Hipótese alternativa


2. Faça o(s) teste(s) de hipóteses adequado(s) para avaliar se existe associação entre hábitos saudáveis e sintomas de depressão.
* ADHERENCE: as categorias do grupo de aderência ao Physical Activity Guideline está associado aos sintomas de depressão.
* ADDSUG: açúcar na alimentação está associado aos sintomas de depressão.
* TOTALDAIRY: laticínios estão associados aos sintomas de depressão.

In [56]:
stat, p = ttest_ind(phq_full[phq_full['phq_grp']=='sem sintomas']['HEI2015_TOTAL_SCORE'], phq_full[phq_full['phq_grp']=='sintomas severos']['HEI2015_TOTAL_SCORE'])
print('HEI2015_TOTAL_SCORE: stat=%.3f, p=%.3f' % (stat, p))
if p>0.05:
    print('Hipótese nula')
else:
     print('Hipótese alternativa')
        
stat, p = ttest_ind(phq_full[phq_full['phq_grp']=='sem sintomas']['PAG_MINW'], phq_full[phq_full['phq_grp']=='sintomas severos']['PAG_MINW'])
print('PAG_MINW: stat=%.3f, p=%.3f' % (stat, p))
if p>0.05:
    print('Hipótese nula')
else:
     print('Hipótese alternativa')
        
stat, p = ttest_ind(phq_full[phq_full['phq_grp']=='sem sintomas']['HEI2015C1_TOTALVEG'], phq_full[phq_full['phq_grp']=='sintomas severos']['HEI2015C1_TOTALVEG'])
print('HEI2015C1_TOTALVEG: stat=%.3f, p=%.3f' % (stat, p))
if p>0.05:
    print('Hipótese nula')
else:
     print('Hipótese alternativa')

stat, p = ttest_ind(phq_full[phq_full['phq_grp']=='sem sintomas']['HEI2015C3_TOTALFRUIT'], phq_full[phq_full['phq_grp']=='sintomas severos']['HEI2015C3_TOTALFRUIT'])
print('HEI2015C3_TOTALFRUIT: stat=%.3f, p=%.3f' % (stat, p))
if p>0.05:
    print('Hipótese nula')
else:
     print('Hipótese alternativa')

stat, p = ttest_ind(phq_full[phq_full['phq_grp']=='sem sintomas']['HEI2015C7_TOTPROT'], phq_full[phq_full['phq_grp']=='sintomas severos']['HEI2015C7_TOTPROT'])
print('HEI2015C7_TOTPROT: stat=%.3f, p=%.3f' % (stat, p))
if p>0.05:
    print('Hipótese nula')
else:
     print('Hipótese alternativa')

stat, p = ttest_ind(phq_full[phq_full['phq_grp']=='sem sintomas']['HEI2015C11_REFINEDGRAIN'], phq_full[phq_full['phq_grp']=='sintomas severos']['HEI2015C11_REFINEDGRAIN'])
print('HEI2015C11_REFINEDGRAIN: stat=%.3f, p=%.3f' % (stat, p))
if p>0.05:
    print('Hipótese nula')
else:
     print('Hipótese alternativa')

stat, p = ttest_ind(phq_full[phq_full['phq_grp']=='sem sintomas']['HEI2015C13_ADDSUG'], phq_full[phq_full['phq_grp']=='sintomas severos']['HEI2015C13_ADDSUG'])
print('HEI2015C13_ADDSUG: stat=%.3f, p=%.3f' % (stat, p))
if p>0.05:
    print('Hipótese nula')
else:
     print('Hipótese alternativa')

stat, p = ttest_ind(phq_full[phq_full['phq_grp']=='sem sintomas']['HEI2015C12_SFAT'], phq_full[phq_full['phq_grp']=='sintomas severos']['HEI2015C12_SFAT'])
print('HEI2015C12_SFAT: stat=%.3f, p=%.3f' % (stat, p))
if p>0.05:
    print('Hipótese nula')
else:
     print('Hipótese alternativa')
        
stat, p = ttest_ind(phq_full[phq_full['phq_grp']=='sem sintomas']['HEI2015C10_SODIUM'], phq_full[phq_full['phq_grp']=='sintomas severos']['HEI2015C10_SODIUM'])
print('HEI2015C10_SODIUM: stat=%.3f, p=%.3f' % (stat, p))
if p>0.05:
    print('Hipótese nula')
else:
     print('Hipótese alternativa')

stat, p = ttest_ind(phq_full[phq_full['phq_grp']=='sem sintomas']['HEI2015C2_GREEN_AND_BEAN'], phq_full[phq_full['phq_grp']=='sintomas severos']['HEI2015C2_GREEN_AND_BEAN'])
print('HEI2015C2_GREEN_AND_BEAN: stat=%.3f, p=%.3f' % (stat, p))
if p>0.05:
    print('Hipótese nula')
else:
     print('Hipótese alternativa')

stat, p = ttest_ind(phq_full[phq_full['phq_grp']=='sem sintomas']['HEI2015C5_WHOLEGRAIN'], phq_full[phq_full['phq_grp']=='sintomas severos']['HEI2015C5_WHOLEGRAIN'])
print('HEI2015C5_WHOLEGRAIN: stat=%.3f, p=%.3f' % (stat, p))
if p>0.05:
    print('Hipótese nula')
else:
     print('Hipótese alternativa')

stat, p = ttest_ind(phq_full[phq_full['phq_grp']=='sem sintomas']['HEI2015C6_TOTALDAIRY'], phq_full[phq_full['phq_grp']=='sintomas severos']['HEI2015C6_TOTALDAIRY'])
print('HEI2015C6_TOTALDAIRY: stat=%.3f, p=%.3f' % (stat, p))
if p>0.05:
    print('Hipótese nula')
else:
     print('Hipótese alternativa')
        
stat, p = ttest_ind(phq_full[phq_full['phq_grp']=='sem sintomas']['HEI2015C9_FATTYACID'], phq_full[phq_full['phq_grp']=='sintomas severos']['HEI2015C9_FATTYACID'])
print('HEI2015C9_FATTYACID: stat=%.3f, p=%.3f' % (stat, p))
if p>0.05:
    print('Hipótese nula')
else:
     print('Hipótese alternativa')

stat, p = ttest_ind(phq_full[phq_full['phq_grp']=='sem sintomas']['HEI2015C8_SEAPLANT_PROT'], phq_full[phq_full['phq_grp']=='sintomas severos']['HEI2015C8_SEAPLANT_PROT'])
print('HEI2015C8_SEAPLANT_PROT: stat=%.3f, p=%.3f' % (stat, p))
if p>0.05:
    print('Hipótese nula')
else:
     print('Hipótese alternativa')

stat, p = ttest_ind(phq_full[phq_full['phq_grp']=='sem sintomas']['ADHERENCE'], phq_full[phq_full['phq_grp']=='sintomas severos']['ADHERENCE'])
print('ADHERENCE: stat=%.3f, p=%.3f' % (stat, p))
if p>0.05:
    print('Hipótese nula')
else:
     print('Hipótese alternativa')

HEI2015_TOTAL_SCORE: stat=0.900, p=0.368
Hipótese nula
PAG_MINW: stat=1.134, p=0.257
Hipótese nula
HEI2015C1_TOTALVEG: stat=0.271, p=0.786
Hipótese nula
HEI2015C3_TOTALFRUIT: stat=0.998, p=0.318
Hipótese nula
HEI2015C7_TOTPROT: stat=1.024, p=0.306
Hipótese nula
HEI2015C11_REFINEDGRAIN: stat=-1.004, p=0.316
Hipótese nula
HEI2015C13_ADDSUG: stat=2.435, p=0.015
Hipótese alternativa
HEI2015C12_SFAT: stat=-1.196, p=0.232
Hipótese nula
HEI2015C10_SODIUM: stat=-0.970, p=0.332
Hipótese nula
HEI2015C2_GREEN_AND_BEAN: stat=0.042, p=0.967
Hipótese nula
HEI2015C5_WHOLEGRAIN: stat=1.620, p=0.105
Hipótese nula
HEI2015C6_TOTALDAIRY: stat=2.219, p=0.027
Hipótese alternativa
HEI2015C9_FATTYACID: stat=-0.479, p=0.632
Hipótese nula
HEI2015C8_SEAPLANT_PROT: stat=0.933, p=0.351
Hipótese nula
ADHERENCE: stat=3.188, p=0.001
Hipótese alternativa


Modelo de regressão linear

In [57]:
import statsmodels.api as sm
import statsmodels.formula.api as smf

In [58]:
phq_full.columns

Index(['ID', 'Q01', 'Q02', 'Q03', 'Q04', 'Q05', 'Q06', 'Q07', 'Q08', 'Q09',
       'Gênero', 'Idade', 'Etnia', 'Escolaridade', 'Renda_familiar',
       'Categoria_idade', 'PAG_MINW', 'ADHERENCE', 'HEI2015C1_TOTALVEG',
       'HEI2015C2_GREEN_AND_BEAN', 'HEI2015C3_TOTALFRUIT',
       'HEI2015C4_WHOLEFRUIT', 'HEI2015C5_WHOLEGRAIN', 'HEI2015C6_TOTALDAIRY',
       'HEI2015C7_TOTPROT', 'HEI2015C8_SEAPLANT_PROT', 'HEI2015C9_FATTYACID',
       'HEI2015C10_SODIUM', 'HEI2015C11_REFINEDGRAIN', 'HEI2015C12_SFAT',
       'HEI2015C13_ADDSUG', 'HEI2015_TOTAL_SCORE', 'phq9', 'phq_grp',
       'Cat_idade_int', 'CAT_TOTAL_SCORE'],
      dtype='object')

 ### 6 - Aspectos da análise

1. Qual o tipo de estudo está sendo empregado pelo NHANES? experimental ou observacional?
* Observacional, os pesquisadores não podem forçar as pessoas do estudo a ingerirem certos alimentos e fazerem uma quantidade de exercício físico por semana.
* Vantagem:
* Alta validade externa (cenário mais próximo da vida real)
* Desvantagens:
* Em geral, não permitem interpretação causal
* A interpretação causal requer suposições fortes

2. Discuta as possíveis fontes de viés presentes na análise.
* Viés de confundimento: o efeito da variável explicativa se mistura aos efeitos de variáveis confundidoras.
    * Gênero feminino está associado aos maiores sintomas de depressão?
        * Variável confundidora: pode ser TPM, depressão pós-parto ou dupla jornada de trabalho.
    * Baixa escolaridade está associada aos maiores sintomas de depressão?
        * Variável confundidora: problemas familiares, bullying, violência doméstica, pobreza extrema, envolvimento com atividades ilícitas, entre outras.
    * A ingestão de laticínios está associada aos maiores sintomas de depressão?
        * Variável confundidora: intolerância à lactose.
    * A ingestão de açúcares está associada aos maiores sintomas de depressão?
        * Variável confundidora: falta de sono, estresse na vida pessoal ou profissional.
    * A menor quantidade de minutos que alguém faz exercícios físicos por semana está associada aos maiores sintomas de depressão?
        * Variável confundidora: estresse na vida pessoal ou profissional, falta de sono, entre outros.

        



3. A partir da análise realizada, podemos afirmar que hábitos saudáveis possuem um efeito causal na prevenção de depressão?
* Não é possível estabelecer uma relação causal entre hábitos saudáveis e sintomas de depressão, porque muitas variáveis confundidoras não foram coletadas.

4. Quais são as fraquezas das análises realizadas? O que pode ser feito para melhorar?
* As fraquezas são os dados insuficientes para estabelecer uma relação causal entre hábitos saudáveis e sintomas de depressão.
* Poucas inferências podem ser feitas a partir de uma grande quantidade de dados.
* Muitas varáveis confundidoras.
* O que pode ser melhorado:
* 1. Controlar (incluir na análise) variáveis confundidoras conhecidas. Iremos supor que o efeito de confundimento de variáveis não coletadas é mínimo
* 2. Matching: selecionar o grupo controle de acordo com características do grupo de exposição (ex.: idade, gênero, localização, etc.)

5. Quais outras variáveis/informações poderiam ter sido coletadas para esta análise?
* A pesquisa poderia adicionar perguntas sobre a qualidade do sono, frequência de estresse no trabalho e na vida pessoal, carga horária juntando tarefas domésticas e trabalho profissional e quais são as intolerâncias e alergias alimentícias.